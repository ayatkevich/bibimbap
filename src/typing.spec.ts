import path from 'path';
import fs from 'fs';
import ts from 'typescript';

test('check typings', () => {
  const program = ts.createProgram(
    fs
      .readdirSync('./src/typing-tests')
      .map(file => path.join('./src/typing-tests', file)),
    {
      baseUrl: './src',
      declaration: false,
      noEmit: true,
      downlevelIteration: true,
      esModuleInterop: true,
      experimentalDecorators: true,
      importHelpers: false,
      inlineSourceMap: true,
      lib: ['lib.es2015.d.ts', 'lib.es2016.array.include.d.ts'],
      module: ts.ModuleKind.CommonJS,
      moduleResolution: ts.ModuleResolutionKind.NodeJs,
      noEmitOnError: true,
      noFallthroughCasesInSwitch: true,
      noImplicitAny: true,
      noImplicitReturns: true,
      noUnusedLocals: true,
      noUnusedParameters: true,
      removeComments: true,
      resolveJsonModule: true,
      skipLibCheck: true,
      sourceMap: false,
      strict: true,
      target: ts.ScriptTarget.ESNext,
      types: ['jest', 'node']
    }
  );
  const emitResult = program.emit();

  const allDiagnostics = ts
    .getPreEmitDiagnostics(program)
    .concat(emitResult.diagnostics);

  for (const diagnostic of allDiagnostics) {
    const message = ts.flattenDiagnosticMessageText(
      diagnostic.messageText,
      '\n'
    );
    if (diagnostic.file) {
      expect(message).toMatchSnapshot(
        `${diagnostic.file.fileName}`
      );
    } else {

      expect(message).toMatchSnapshot();
    }
  }
});
