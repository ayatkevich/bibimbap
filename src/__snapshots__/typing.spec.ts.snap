// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check typings: src/typing-tests/jsql.ts (14,24) 1`] = `
"Argument of type '{}' is not assignable to parameter of type 'TableProperties<Table<\\"TestTable\\", ColumnFree<\\"isNullable\\", BooleanConstructor, {}, true> | ColumnFree<\\"withDefault\\", NumberConstructor, number, boolean | undefined> | ColumnFree<\\"withDefaultAndNullable\\", StringConstructor, string, true> | ColumnFree<...>>>'.
  Type '{}' is not assignable to type '{ required: string; }'.
    Property 'required' is missing in type '{}'."
`;

exports[`check typings: src/typing-tests/jsql.ts (16,24) 1`] = `
"Argument of type '{ isNullable: false; }' is not assignable to parameter of type 'TableProperties<Table<\\"TestTable\\", ColumnFree<\\"isNullable\\", BooleanConstructor, {}, true> | ColumnFree<\\"withDefault\\", NumberConstructor, number, boolean | undefined> | ColumnFree<\\"withDefaultAndNullable\\", StringConstructor, string, true> | ColumnFree<...>>>'.
  Type '{ isNullable: false; }' is not assignable to type '{ required: string; }'.
    Property 'required' is missing in type '{ isNullable: false; }'."
`;
